### PR TITLE
Validate form on init of step 3

### PR DESCRIPTION
**Pull request by kravets-levko**
1. Trigger validation on init of step 3 (if there are some filled fields).
2. Debounce validation (immediate only on init of step 3).

### CODE REVIEW

File: app/front/scripts/controllers/provide-metadata.js [hunk 1]
```diff
@@ -25,36 +32,41 @@ angular.module('Application')
           end: fiscalPeriod ? fiscalPeriod.end : ''
         };
 
-        $scope.$watch('attributes.title', function() {
```
Comment (akariv) [inline app/front/scripts/controllers/provide-metadata.js:28]: Why were these removed?

Comment (kravets-levko) [reply app/front/scripts/controllers/provide-metadata.js:28]: Because it is redundant: we already have the same watcher (several lines below) that deeply watches entire `attributes` property.

Comment (akariv) [reply app/front/scripts/controllers/provide-metadata.js:28]: But in that watch we check `if ((newValue === oldValue))`, which basically means that if the object is the same one (regardless of property changes) then we won't do anything, no?

Comment (kravets-levko) [reply app/front/scripts/controllers/provide-metadata.js:28]: It's one of the strange things in angular. Look at $watch prototype: `$scope.$watch(watchExpression, listener, [objectEquality = False]);`. Listener is a function that takes two parameters: new and old value of property. Listener always is called only if property changes, so old value is newer the same as new value; `objectEquality` is used to specify how to compare objects: as references or deeply - by default Angular will compare objects as references. And there is the only case when `listener` may be called with equal arguments - it's a first time after registering watcher - in this case both new and old values are equal to `undefined` - so `newValue !== oldValue` is just a recommended way to ignore that first call. And if you look at watcher for `attributes` - there is last parameter set to true.

Comment (akariv) [reply app/front/scripts/controllers/provide-metadata.js:28]: Okay - I just realised that specifying `true` in `objectEquality` actually means that angular will copy the entire object every time something changes.
If the model is complicated it could become quite intensive, no?

Comment (kravets-levko) [reply app/front/scripts/controllers/provide-metadata.js:28]: Yes, definitely - it could. But in this case there are no too much data - `attributes` object is almost plain and has no nested objects, so it's quite fast.

Comment (akariv) [reply app/front/scripts/controllers/provide-metadata.js:28]: 👍
